### PR TITLE
manager: allow assignment of properties on target/swap/device units 

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/core/dbus-unit.c [hunk 1]
```diff
@@ -2437,10 +2437,6 @@ int bus_unit_set_properties(
                 if (r < 0)
                         return r;
 
-                if (!UNIT_VTABLE(u)->bus_set_property)
```
Comment (yuwata) [inline src/core/dbus-unit.c:2440]: Maybe, it is safer to add assertion for `bus_set_property` below?

Comment (yuwata) [inline src/core/dbus-unit.c:2440]: Hm? Wait, `bus_set_property` is not set for .target and .device??

Comment (yuwata) [inline src/core/dbus-unit.c:2440]: Yeah, it seems necessary to conditionalize the call `bus_set_property()` below.

**Comment by mrc0mmand** [comment]
Interesting, `dfuzzer` manages to crash systemd with this patch:

```
[   60.613099] testsuite-21.sh[1433]: Object: /org/freedesktop/systemd1/unit/dev_2dttyS21_2edevice
[   60.622679] testsuite-21.sh[1433]:  Interface: org.freedesktop.DBus.Peer
[   60.641949] testsuite-21.sh[1433]:  Interface: org.freedesktop.DBus.Introspectable
[   60.666977] testsuite-21.sh[1433]:  Interface: org.freedesktop.DBus.Properties
[   60.964477] testsuite-21.sh[1433]:  Interface: org.freedesktop.systemd1.Device
[   60.978448] testsuite-21.sh[1433]:  Interface: org.freedesktop.systemd1.Unit
[   61.653830] systemd[1]: Caught <SEGV> from unknown sender process.
[   61.654883] audit[1755]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1755 comm="systemd" exe="/usr/lib/systemd/systemd" sig=11 res=1
[   61.678821] systemd-coredump[1756]: Due to PID 1 having crashed coredump collection will now be turned off.
[   64.402246] systemd-journald[819]: Failed to write entry to /var/log/journal/927634ea688e46418f8f163a978a7185/system.journal (44 items, 146722818 bytes) despite vacuuming, ignoring: Argument list too long
[   64.357112] audit: BPF prog-id=0 op=UNLOAD
[   64.357297] systemd[1]: Caught <SEGV>, dumped core as pid 1755.
[   64.357791] testsuite-21.sh[1433]:   FAIL [M] SetProperties - unexpected response
[   64.357791] testsuite-21.sh[1433]:    on input:
[   64.357791] testsuite-21.sh[1433]:    -- Signature: (ba(sv))
[   64.357791] testsuite-21.sh[1433]:    -- Value: (true, [('%s%s%s%s%s%s%s%s%s%n%s%n%n%n%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s', <('%s%s%s%s%s%s%s%s%s%n%s%n%n%n%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s',)>), ('%s%s%s%s%s%s%s%s%s%n%s%n%n%n%s%s%s>
[   64.357791] testsuite-21.sh[1433]:    reproducer: dfuzzer -v -n org.freedesktop.systemd1 -o /org/freedesktop/systemd1/unit/dev_2dttyS21_2edevice -i org.freedesktop.systemd1.Unit -t SetProperties -b 10000
[   64.361396] systemd[1]: Freezing execution.
[   64.361587] systemd-oomd[1754]: Failed to connect to /run/systemd/io.system.ManagedOOM: Connection refused
[   64.361807] systemd-oomd[1754]: Failed to acquire varlink connection: Connection refused
[   64.361911] systemd-oomd[1754]: Event loop failed: Connection refused
[   87.803595] dbus-daemon[893]: [system] Failed to activate service 'org.freedesktop.systemd1': timed out (service_start_timeout=25000ms)
[   87.804639] testsuite-21.sh[1433]: Error: Unable to create proxy for bus name 'org.freedesktop.systemd1'.
[   87.805243] testsuite-21.sh[1433]: Exit status: 1

```

File: src/core/dbus-unit.c (hunk 1 above)
Comment (keszybz) [reply src/core/dbus-unit.c:2440]: Duh, that was obvious.

Comment (keszybz) [reply src/core/dbus-unit.c:2440]: Yeah, because they don't have any type-specific properties. So it should be OK with the updated version.

**Review by yuwata — APPROVED** [review]

**Pull request merged**